src/rest-server/src/utils/frameworkConverter.js

```diff
@@ -379,7 +466,143 @@ const convertTaskDetail = async (
   };
 };
 
+const convertTaskAttempt = async (
+  logPathInfix,
+  ports,
+  taskStatus,
+  taskRoleName,
+  userName,
+  attemptStatus,
+) => {
+  // get containerPrts
+  const containerPorts = getContainerPorts(
+    ports,
+    taskStatus.index,
+    taskStatus.attemptStatus.podUID,
+  );
+  // get affinity group name
+  const affinityGroupName = `default/${taskStatus.attemptStatus.podName}`;
+  // get container gpus
+  const containerGpus = await getContainerGpus(
+    true,
+    taskStatus.attemptStatus.podName,
+  );
+
+  const completionStatus = taskStatus.attemptStatus.completionStatus;
+  const diagnostics = completionStatus ? completionStatus.diagnostics : null;
+  const exitDiagnostics = generateExitDiagnostics(diagnostics);
+
+  return {
+    attemptId: attemptStatus.id,
+    attemptState: convertAttemptState(
+      taskStatus.state,
+      completionStatus ? completionStatus.code : null,
+    ),
+    currentAttemptLaunchedTime:
+      new Date(attemptStatus.runTime || attemptStatus.startTime).getTime() ||
+      null,
+    currentAttemptCompletedTime:
+      new Date(attemptStatus.completionTime).getTime() || null,
+    containerId: attemptStatus.podUID,
+    containerIp: attemptStatus.podHostIP,
+    containerNodeName: attemptStatus.podNodeName,
+    // Job level info
+    containerPorts,
+    containerGpus,
+    containerLog: `http://${attemptStatus.podHostIP}:${process.env.LOG_MANAGER_PORT}/log-manager/tail/${userName}/${logPathInfix}/${taskRoleName}/${attemptStatus.podUID}/`,
+    containerExitCode: completionStatus ? completionStatus.code : null,
+    containerExitSpec: completionStatus
+      ? generateExitSpec(completionStatus.code)
+      : generateExitSpec(null),
+    containerExitDiagnostics: exitDiagnostics
+      ? exitDiagnostics.diagnosticsSummary
+      : null,
+    ...(launcherConfig.enabledHived && {
+      hived: {
+        affinityGroupName,
+        lazyPreempted: null,
+        lazyPreemptionStatus: null,
+      },
+    }),
+  };
+};
+
+const convertToTaskDetail = async (
+  attemptFramework,
+  taskRoleName,
+  taskStatus,
+  taskHistories,
+) => {
+  const lastTaskAttemptStatus = taskStatus.attemptStatus;
+  const completionStatus = lastTaskAttemptStatus.completionStatus;
+  const userName = attemptFramework.metadata.labels.userName;
+  const jobName = attemptFramework.metadata.annotations.jobName;
+
+  const taskDetail = {
+    // framework level information
+    frameworkName: userName + '~' + jobName,
```

Comment:
>frameworkName [](start = 4, length = 13)

Remove frameworkName and add userName and jobName?

Reply:
> > frameworkName [](start = 4, length = 13)
> 
> Remove frameworkName and add userName and jobName?
> 
> As we need to and already converted FC framework concepts to PAI job concepts in RestServer

Fixed, thanks